Tests/SwiftParserTest/translated/RecoveryTests.swift

```diff
@@ -809,9 +844,12 @@ final class RecoveryTests: XCTestCase {
       let (efg 1️⃣hij, foobar) = (5, 6)
```

Comment:
Maybe I wasn’t clear in my previous comment. What I thought was that the parser (not `ParseDiagnosticGenerator`) will synthesize a missing comma for this case, ie. parsing its s `(efg, his, foobar)`

If `hij` had a capital `H`, i.e. if the test case was `(efg Hij, foobar)`  the user was more likely to have forgotten the colon because `Hij` is by most conventions a type since it starts with a capital letter, so the parser should synthesize a missing colon and parse it as `(efg: Hij, foobar)`. That way I think we don’t actually need any special logic in `ParseDiagnosticGenerator` (or at least not much).